File: 백준/Silver/4134. 다음 소수/다음 소수.py
import sys


def isPrime(n):
    for i in range(2, int(n ** 0.5) + 1):
        if n % i == 0:
            return False

    return True


N = int(sys.stdin.readline())

for _ in range(N):
    n = int(sys.stdin.readline())

    while True:
        if n == 0 or n == 1:
            print(2)
            break
        else:
            if isPrime(n):
                print(n)
                break
        n += 1

File: 백준/Silver/4134. 다음 소수/다음 소수.java
import java.awt.*;
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.*;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int N = Integer.parseInt(st.nextToken());
        StringBuilder sb = new StringBuilder();

        for (int i = 0; i < N; i++) {
            st = new StringTokenizer(br.readLine());
            long n = Long.parseLong(st.nextToken());
            while (true) {
                if (n == 0 || n == 1) {
                    sb.append(2).append("\n");
                    break;
                } else if (isPrime(n)) {
                    sb.append(n).append("\n");
                    break;
                }
                n += 1;
            }
        }
        System.out.println(sb);
    }

    static boolean isPrime(long n) {
        for (int i = 2; i <= (int) Math.sqrt(n); i++) {
            if (n % i == 0) {
                return false;
            }
        }
        return true;
    }
}